src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -332,11 +326,18 @@ CAC.Control.Directions = (function (_, $, moment, Control, Geocoder, Routing, Ty
         typeaheadFrom.setValue(destinationText);
         typeaheadTo.setValue(originText);
 
-        // set on this object and validate
-        setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
-        setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        // have both origin and destination
+        if (directions.origin && directions.destination) {
+            // set on this object and validate
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.origin) {
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.destination) {
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+        }
 
-        // update the directions for the reverse trip
+        // update the directions for the reverse trip, or clear itineraries
```

Comment:
Should be fixed by #668; see note above. Code sets 'hidden' class on sidebar, but styling for 'hidden' got removed when we stripped out Bootstrap.